tests/e2e/specs/dashboard/dashboard-noscript.test.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * WordPress dependencies
+ */
+import { visitAdminPage } from '@wordpress/e2e-test-utils';
+
+describe( 'Site Kit noscript notice', () => {
+	beforeEach( async () => {
+		await visitAdminPage( 'admin.php', 'page=googlesitekit-splash' );
+	} );
+
+	describe( 'When Javascript is enabled', () => {
+		it( 'Should not display noscript notice', async () => {
+			const noscript = await page.$( '#wpbody-content' );
+			await expect( noscript ).not.toMatchElement( '.googlesitekit-noscript' );
+		} );
+	} );
+
+	describe( 'When Javascript is disabled', () => {
+		beforeAll( async () => {
+			await page.setJavaScriptEnabled( false );
+		} );
+
+		afterAll( async () => {
+			await page.setJavaScriptEnabled( true );
+		} );
+
+		it( 'Should not display plugin html', async () => {
+			const noscript = await page.$( '#wpbody-content' );
+			await expect( noscript ).not.toMatchElement( '.js-googlesitekit-plugin' );
```

Comment:
This will need updating to not rely on this new class which should be removed.

Also, `noscript` isn't the best name for a reference to `#wpbody-content` 😉